mcs/class/System/Mono.AppleTls/Certificate.cs

```diff
@@ -331,7 +361,67 @@ public void Dispose ()
 
 		protected virtual void Dispose (bool disposing)
 		{
-			if (handle != IntPtr.Zero){
+			if (owner != IntPtr.Zero) {
+				CFObject.CFRelease (owner);
+				owner = handle = IntPtr.Zero;
+			} else if (handle != IntPtr.Zero) {
+				CFObject.CFRelease (handle);
+				handle = IntPtr.Zero;
+			}
+		}
+	}
+
+	class SecAccess : INativeObject, IDisposable {
+		internal IntPtr handle;
+
+		public SecAccess (IntPtr handle, bool owns = false)
+		{
+			this.handle = handle;
+			if (!owns)
+				CFObject.CFRetain (handle);
+		}
+
+		~SecAccess ()
+		{
+			Dispose (false);
+		}
+
+		public IntPtr Handle {
+			get {
+				return handle;
+			}
+		}
+
+		[DllImport (AppleTlsContext.SecurityLibrary)]
+		extern static /* OSStatus */ SecStatusCode SecAccessCreate (/* CFStringRef */ IntPtr descriptor,  /* CFArrayRef */ IntPtr trustedList, /* SecAccessRef _Nullable * */ out IntPtr accessRef);
+
+		public static SecAccess Create (string descriptor)
+		{
+			var descriptorHandle = CFString.Create (descriptor);
+			if (descriptorHandle == null)
+				throw new InvalidOperationException ();
+
+			try {
+				IntPtr accessRef;
+				var result = SecAccessCreate (descriptorHandle.Handle, IntPtr.Zero, out accessRef);
+				if (result != SecStatusCode.Success)
+					throw new InvalidOperationException (result.ToString ());
+
+				return new SecAccess (accessRef, true);
+			} finally {
+				descriptorHandle.Dispose ();
```

Comment:
Bad idea, but we're incorrectly doing it in so many other places.  We'd have to make all those `CFString.Create()` etc. functions throw instead of possibly returning null.